### PR TITLE
use pod-name[.namespace] to replace proxy_name in istioctl usage

### PR DESCRIPTION
**Pull request by clyang82**
Signed-off-by: clyang82 <clyang@cn.ibm.com>

https://github.com/istio/istio/pull/9533 is very good enhancement to have a good user experience to keep `istioctl proxy-status` and `istioctl proxy-config` consistence.

This PR is expose this enhancement in usage to let user know this enhancement.

### CODE REVIEW

File: istioctl/cmd/istioctl/proxyconfig.go [hunk 1]
```diff
@@ -37,25 +37,25 @@ var (
 		Short: "Retrieve information about proxy configuration from Envoy [kube only]",
 		Long:  `A group of commands used to retrieve information about proxy configuration from the Envoy config dump`,
 		Example: `  # Retrieve information about proxy configuration from an Envoy instance.
-  istioctl proxy-config <clusters|listeners|routes|endpoints|bootstrap> <pod-name>`,
+  istioctl proxy-config <clusters|listeners|routes|endpoints|bootstrap> <proxy-name|pod-name>`,
```
Comment (fisherxu) [inline istioctl/cmd/istioctl/proxyconfig.go]: we care the pod name here, so what does the proxy-name mean?

Comment (clyang82) [reply istioctl/cmd/istioctl/proxyconfig.go]: proxy-name is original concept in proxy-status. for example:
proxy-name: details-v1-64f56d694d-6rh5s.default while pod name is details-v1-64f56d694d-6rh5s

Comment (ymesika) [reply istioctl/cmd/istioctl/proxyconfig.go]: @clyang82 It is still unclear to me what's the proxy-name. Is it just the pod+ns? If that's the case then you can add the `-n` flag today.

Comment (clyang82) [reply istioctl/cmd/istioctl/proxyconfig.go]: @ymesika please refer to this issue - https://github.com/istio/istio/pull/9533. in order to keep consistence, add support pod-name and proxy-name for both `proxy-config` and `proxy-status`. My PR is to expose them to tell user we support both.

Comment (ymesika) [reply istioctl/cmd/istioctl/proxyconfig.go]: I see. So it has nothing to do with the proxy, right? I think the `proxy-name` name is confusing and doesn't tell user what is expected.
What do you think of `<pod-name[.namespace]>` ?

Comment (costinm) [reply istioctl/cmd/istioctl/proxyconfig.go]: +1 on pod-name[.namespace]

Comment (clyang82) [reply istioctl/cmd/istioctl/proxyconfig.go]: @ymesika & @costinm Thanks for your comments. How about the `proxy-status`. Here is help output:
```
istioctl proxy-status --help

Retrieves last sent and last acknowledged xDS sync from Pilot to each Envoy in the mesh

Usage:
  istioctl proxy-status [<proxy-name>] [flags]
```

and the output have `PROXY`
```
PROXY                                                  CDS        LDS        EDS               RDS          PILOT                            VERSION
details-v1-64f56d694d-6rh5s.default                    SYNCED     SYNCED     SYNCED (50%)      SYNCED       istio-pilot-7899d4d478-f9925     1.1.0
```
We should change them as well. `PROXY` --> `POD.NAMESPACE`?

Comment (ymesika) [reply istioctl/cmd/istioctl/proxyconfig.go]: I think it should be `istioctl proxy-status [<pod-name>.<namespace>] [flags]`.
Just tried without the ns and it requires the full name.

Comment (clyang82) [reply istioctl/cmd/istioctl/proxyconfig.go]: Do you mean `istioctl proxy-status [<pod-name[.namespace]>] [flags]`?

How about PROXY --> POD in output of `istioctl proxy-status`?

Comment (fisherxu) [reply istioctl/cmd/istioctl/proxyconfig.go]: I think it's not only `[<pod-name[.namespace]>] `, `-n namespace` also works.

Comment (clyang82) [reply istioctl/cmd/istioctl/proxyconfig.go]: @fisherxu In my understand, `[<pod-name[.namespace]>]` means both `pod-name.namespace` and `pod-name -n namespace` works.

Comment (fisherxu) [reply istioctl/cmd/istioctl/proxyconfig.go]: Make sense to me. Thanks :) @clyang82 Could you squash the commits?

Comment (clyang82) [reply istioctl/cmd/istioctl/proxyconfig.go]: @fisherxu squashed. Thanks 👍

**Comment by clyang82** [comment]
@fisherxu Can you help to merge this pull request if you have no further comments? Thanks.

**Comment by clyang82** [comment]
@fisherxu  👍 Thank you

**Comment by fisherxu** [comment]
@clyang82 NP ;-)

**Pull request merged**